docs/proposals/20221003-In-place-propagation-of-Kubernetes-objects-only-changes.md

```diff
@@ -0,0 +1,321 @@
+(for Clusters with ClusterClass).
+
+As a cluster admin/user, I would like a declarative and secure means by which to assign roles to my nodes via KubeadmControlPlane and
+MachineDeployments (for Clusters without ClusterClass).
+
+#### Story 2
+
+As a cluster admin/user, I would like to change labels or annotations on Machines without triggering Machine rollouts.
+
+#### Story 3
+
+As a cluster admin/user, I would like to change nodeDrainTimeout on Machines without triggering Machine rollouts.
+
+#### Story 4
+
+As a cluster admin/user, I would like to set autoscaler labels for MachineDeployments by changing Cluster topology metadata
+(for Clusters with ClusterClass).
+
+### Implementation Details/Notes/Constraints
+
+### Metadata propagation
+
+The following schema represent how metadata propagation works today (also documented in [book](https://cluster-api.sigs.k8s.io/developer/architecture/controllers/metadata-propagation.html)).
+
+![Figure 1](./images/in-place-propagation/current-state.png)
+
+With this proposal we are suggesting to improve metadata propagation as described in the following schema:
+
+![Figure 2](./images/in-place-propagation/proposed-changes.png)
+
+Following paragraphs provide more details about the proposed changes.
+
+#### 1. Label Sync Between Machine and underlying Kubernetes Nodes
+
+As discussed in [Label Sync Between Machine and underlying Kubernetes Nodes](./20220927-label-sync-between-machine-and-nodes.md) we are propagating only
+labels with a well-known prefix or a well-known domain from the Machine to the corresponding Kubernetes Node.
+
+#### 2. Labels/Annotations always reconciled
+
+All the labels/annotations previously set only on creation are now going to be always reconciled;
+in order to prevent unnecessary rollouts, metadata propagation should happen in-place;
+see [in-place propagation](#in-place-propagation) down in this document for more details. 
+
+Note: As of today the topology controller already propagates ClusterClass and Cluster topology metadata changes in-place when possible
+in order to avoid unnecessary template rotation with the consequent Machine rollout; we do not foresee changes to this logic.
+
+#### 3. and 4. Set top level labels/annotations for ControlPlane and MachineDeployment created from a ClusterClass
+
+labels and annotation from ClusterClass and Cluster.topology are going to be propagated to top-level level labels in
+ControlPlane and MachineDeployment.
+
+This address [Labels and annotations for MachineDeployments and KubeadmControlPlane created by topology controller](https://github.com/kubernetes-sigs/cluster-api/issues/7006).
+
+Note: The proposed solution avoids to add additional metadata fields in ClusterClass and Cluster.topology, but
+this has the disadvantage that it is not possible to differentiate top-level labels/annotations from Machines,
+but given the discussion on the above issue this isn't a issue.
```

Comment:
```suggestion
but given the discussion on the above issue this isn't a requirement.
```
(nit: just to avoid repetition of issue, alternatively `a issue` => `an issue`)